src/accelerate/utils/dataclasses.py

```diff
@@ -549,6 +551,40 @@ def zero3_init_context_manager(self, enable=False):
             self.dschf = None
             self.set_deepspeed_weakref()
 
+    def _deepspeed_config_checks(self):
+        env_variable_names_to_ignore = [
+            "ACCELERATE_GRADIENT_ACCUMULATION_STEPS",
+            "ACCELERATE_GRADIENT_CLIPPING",
+            "ACCELERATE_DEEPSPEED_ZERO_STAGE",
+            "ACCELERATE_DEEPSPEED_OFFLOAD_OPTIMIZER_DEVICE",
+            "ACCELERATE_DEEPSPEED_OFFLOAD_PARAM_DEVICE",
+            "ACCELERATE_DEEPSPEED_ZERO3_SAVE_16BIT_MODEL",
+            "ACCELERATE_MIXED_PRECISION",
+        ]
+        duplicate_values_flag = False
+        for name in env_variable_names_to_ignore:
+            if name != "ACCELERATE_MIXED_PRECISION" and os.environ.get(name, None) is not None:
+                duplicate_values_flag = True
+            elif name == "ACCELERATE_MIXED_PRECISION" and os.environ.get(name, "no") != "no":
+                duplicate_values_flag = True
+            if duplicate_values_flag:
+                break
+
+        if duplicate_values_flag:
+            env_variable_names_to_ignore = [
+                name.replace("ACCELERATE_", "").replace("DEEPSPEED_", "").lower()
+                for name in env_variable_names_to_ignore
+            ]
+            raise ValueError(
+                f"When using `deepspeed_config_file`, the following accelerate config variables will be ignored: {env_variable_names_to_ignore}.\n"
+                "Please specify them appropriately in the DeepSpeed config file.\n"
+                "If you are using an accelerate config file, set `mixed_precision=no` "
```

Comment:
May I ask to resolve this config as well. Did I understand it right that here you suggest to the user to set `mixed_precision=no` even if they use mixed precision in ds_config.json? If so this now is worse than before, as now you're proposing to force 2 totally opposite settings in the same config entry.

Unless I'm misunderstanding the wording that is.

Reply:
Done. I've simplified the error message wherein user can just remove all the ambiguous entries from  accelerate config file and not specify them in launch command

Reply:
to focus we are dealing with `mixed_precision` duplicity here:

I've just tried:

```
deepspeed_config:
  deepspeed_multinode_launcher: standard
  deepspeed_config_file: ./configs/vopt-large-z3/ds_config.json
  zero3_init_flag: true
distributed_type: DEEPSPEED
mixed_precision: 'no'
```

```
{
    "fp16": {
        "enabled": true,
        [...]
```
and it doesn't assert.

Reply:
(incidentally ``mixed_precision`: fp16|bf16|'no'` is odd - why do different values don't follow the same style - i.e 'no' in quotes but not the other values)

Reply:
Hello @stas, this case can't be helped as 'no' is the default value and if one doesn't specify the entry in accelerate config file, it will get default value of 'no' and thereby we have no way to check if user has given the default in config file or was it set to default when reading the config file. It would mean a lot of code rewriting as `mixed_precision` is used in various other non DeepSpeed parts.

The default value of `mixed_precision` would be overriden by that in `deepspeed_config_file`. Now, if you answer the questionnaire via `accelerate config`, the entry of `mixed_precision` won't be there when using `deepspeed_config_file`. 


@sgugger, what are your thoughts around this.

Reply:
understood! in which case I think this should work:

1. if the `mixed_precision` config isn't there all is good
2. if it is there and ds_config is on it must match  ds_config